apps/web/src/services/onboard/ledger-module.ts

```diff
@@ -0,0 +1,388 @@
+        }
+
+        // Sets the current account and emits the accountsChanged event
+        function setCurrentAccount(account: Account): void {
+          currentAccount = account
+          eventEmitter.emit('accountsChanged', [currentAccount.address])
+        }
+
+        // Clears the current account and emits the accountsChanged event
+        function clearCurrentAccount(): void {
+          currentAccount = null
+          eventEmitter.emit('accountsChanged', [])
+        }
+
+        // Clears the current chain and emits the chainChanged event
+        function clearCurrentChain(): void {
+          currentChain = DEFAULT_CHAIN
+          eventEmitter.emit('chainChanged', currentChain.id)
+        }
+
+        // Gets the asserted derivation path from the current account
+        function getAssertedDerivationPath(): DerivationPath {
+          if (!currentAccount?.derivationPath) {
+            throw new ProviderRpcError({
+              code: -32000, // Method handler crashed
+              message: 'No derivation path found',
+            })
+          }
+          return currentAccount.derivationPath
+        }
+
+        /* -------------------------------------------------------------------------- */
+        /*                              EIP-1193 provider                             */
+        /* -------------------------------------------------------------------------- */
+
+        const eip1193Provider = createEIP1193Provider(
+          getHardwareWalletProvider(() => {
+            const rpcUrl = currentChain.rpcUrl
+            if (!rpcUrl) {
+              throw new ProviderRpcError({
+                code: ProviderRpcErrorCode.UNRECOGNIZED_CHAIN_ID,
+                message: `No RPC found for chain ID: ${currentChain.id}`,
+              })
+            }
+            return rpcUrl
+          }),
+          {
+            eth_requestAccounts: async () => {
+              const accounts = await getAccounts()
+              return [accounts[0].address]
+            },
+            eth_selectAccounts: async () => {
+              const accounts = await getAccounts()
+              return accounts.map((account) => account.address)
+            },
+            eth_accounts: async () => {
+              if (!currentAccount) {
+                return []
+              }
+              return [currentAccount.address]
+            },
+            eth_chainId: async () => {
+              return currentChain.id
+            },
+            eth_signTransaction: async (args) => {
+              const txParams = args.params[0]
+
+              const gasLimit = txParams.gas ?? txParams.gasLimit
+              const nonce =
+                txParams.nonce ??
+                // Safe creation does not provide nonce
+                ((await eip1193Provider.request({
+                  method: 'eth_getTransactionCount',
+                  params: [currentAccount!.address, 'latest'],
```

Comment:
Would `pending` be better here in case you still got pending transactions?
I think `latest` only considers already mined blocks.

```suggestion
                  params: [currentAccount!.address, 'pending'],
```

Reply:
Updated in 9f6c12e9509068ac6be90ee199fb54de3f5ba022.